src/services/webPlatformUtils.service.ts

```diff
@@ -224,24 +224,61 @@ export class WebPlatformUtilsService implements PlatformUtilsService {
 
     async showPasswordDialog(title: string, body: string, passwordValidation: (value: string) => Promise<boolean>):
         Promise<boolean> {
+
+        const html = `
```

Comment:
It really feels like we need look at switching to using templates or components to pop into these dialogs vs. embedding HTML in strings and logic around where the dialog is shown (vs. encapsulating the dialog into its own component w/ logic and display).

Reply:
Yea I felt that way as well. I will spend a bit of time looking into using the `repromptService` to show a dialog instead of using SweetAlert.

Reply:
Decided to merge this for now, and postpone move to modals.